src/main/java/uk/ac/ebi/ampt2d/accession/common/persistence/DatabaseService.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.common.persistence;
+
+import uk.ac.ebi.ampt2d.accession.common.generators.ModelHashAccession;
+import uk.ac.ebi.ampt2d.accession.common.accessioning.SaveResponse;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public interface DatabaseService<MODEL, HASH, ACCESSION> {
+
+    Map<ACCESSION, MODEL> findAllAccessionByMessageHash(Collection<HASH> messageHashes);
```

Comment:
Remove `message` from the method and parameter names in this class, it only makes them more confusing.